components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/internal/dao/impl/WebhookManagementDAOFacade.java

```diff
@@ -161,7 +161,7 @@ public void updateWebhook(Webhook webhook, int tenantId) throws WebhookMgtExcept
                 if (!existingWebhook.getEndpoint().equals(webhook.getEndpoint()) &&
                         isWebhookEndpointExists(webhook.getEndpoint(), tenantId)) {
                     throw WebhookManagementExceptionHandler.handleClientException(
-                            ErrorMessage.ERROR_CODE_WEBHOOK_ALREADY_EXISTS);
+                            ErrorMessage.ERROR_CODE_WEBHOOK_ENDPOINT_ALREADY_EXISTS, webhook.getEndpoint());
                 }
```

Comment:
This validation should go to service layer. Also, if secret is not present it should populate the secret from the existin webhook at update. In the PUT secret should be optional to update (we are sligtly deviating from the PUT)

In addition to this I noted that the service layer does not return the secret. Ideally service layer should return. It's the API layer that should not return the secret externally.

We will also need to store the secret in the secret vault as HMAC signing happens at websubhub.

Reply:
Fixed

Reply:
Tracked the imprvement for storing the secret in the secret vault in here https://github.com/wso2/product-is/issues/24084